### PR TITLE
bug 1700903: standardize resource names

### PR DESCRIPTION
**Pull request by ericavonb**
Remove "openshift-" prefix from the resource names for auth components

### CODE REVIEW

File: pkg/operator2/operator.go [hunk 1]
```diff
@@ -41,13 +41,15 @@ import (
 var deploymentVersionHashKey = operatorv1.GroupName + "/rvs-hash"
 
 const (
-	clusterOperatorName = "authentication"
-	targetName          = "openshift-authentication"
-	targetNameOperator  = "openshift-authentication-operator"
-	globalConfigName    = "cluster"
+	clusterOperatorName     = "authentication"
+	targetName              = "oauth-server"
```
Comment (enj) [inline pkg/operator2/operator.go]: Why `oauth-server` vs `integrated-oauth-server` to match existing APIs like authentication `spec.type`?

**Comment by enj** [comment]
/hold

Need some reasoning on the name choice.

File: pkg/operator2/operator.go [hunk 2]
```diff
@@ -41,13 +41,15 @@ import (
...
+	targetNamespace         = "openshift-authentication"
+	targetNameOperator      = "authentication-operator"
+	targetNamespaceOperator = "authentication-operator"
```
Comment (enj) [inline pkg/operator2/operator.go]: For any const that you change, create a new variable name to make it clear what is getting updated.

Also, this const value is wrong.

**Comment by ericavonb** [comment]
/retest
gotta love getting our own flake

**Pull request merged**